corehq/apps/data_cleaning/tests/test_utils_cases.py

```diff
@@ -3,498 +3,702 @@
+            data_type=CaseProperty.DataType.GPS,
+            index=1,
+        )
+        details = get_property_details_from_data_dictionary(self.domain, self.case_type)
+        expected_result = {
+            'plant_location': {
+                'label': 'Plant Location',
+                'data_type': DataType.GPS,
+                'prop_id': 'plant_location',
+                'is_editable': True,
+                'options': [],
+            },
+        }
+        assert details == expected_result
+
+    @privilege_enabled(privileges.DATA_DICTIONARY)
+    def test_no_properties(self):
+        details = get_property_details_from_data_dictionary(self.domain, self.case_type)
+        expected_result = {}
+        assert details == expected_result
+
+    def test_no_privilege(self):
+        CaseProperty.objects.create(
+            case_type=self.case_type_dd,
+            name='plant_location',
+            label='Plant Location',
+            data_type=CaseProperty.DataType.GPS,
+            index=1,
+        )
+        CaseProperty.objects.create(
+            case_type=self.case_type_dd,
+            name='undefined_property',
+            label='Undefined Property',
+            data_type=CaseProperty.DataType.UNDEFINED,
+            index=2,
+        )
+        details = get_property_details_from_data_dictionary(self.domain, self.case_type)
+        expected_result = {}
+        assert details == expected_result
+
+
+def get_mock_all_case_properties_by_domain(result_properties):
+    @quickcache(vary_on=['domain', 'include_parent_properties', 'exclude_deprecated_properties'])
+    def _mock_all_case_properties_by_domain(
+        domain, include_parent_properties=True, exclude_deprecated_properties=True
+    ):
+        return result_properties
+
+    return _mock_all_case_properties_by_domain
+
+
+@privilege_enabled(privileges.DATA_DICTIONARY)
+@disable_quickcache
+class GetCasePropertyDetailsTest(BaseCaseUtilsTest):
+    case_type = 'plant'
 
     @mock.patch(
         'corehq.apps.data_cleaning.utils.cases.all_case_properties_by_domain',
-        _mock_all_case_properties_by_domain,
+        get_mock_all_case_properties_by_domain(
+            {
+                case_type: [],
+                'other': [],
+            }
+        ),
     )
-    @privilege_enabled(privileges.DATA_DICTIONARY)
-    def test_expected_format_with_dd_entries(self):
-        clear_caches_case_data_cleaning(self.domain, 'plant')
-        self._generate_data_dictionary_entries()
-        details = get_case_property_details(
-            self.domain,
-            'plant',
+    def test_text_system_property(self):
+        details = get_case_property_details(self.domain, self.case_type)
+        expected_result = {
+            'label': 'Owner',
+            'data_type': DataType.TEXT,
+            'prop_id': 'owner_name',
+            'is_editable': False,
+            'options': None,
+        }
+        assert details['owner_name'] == expected_result
```

Comment:
There's a lot of duplication between a tests like `test_test_system_property` and `test_date_system_property`, because they are testing both the output format as well as the type conversion. Especially because there are so many type conversion tests, if the format were to change in the future, it would be annoying to have to update all of these tests for essentially an unrelated change. I'd recommend creating a single generic "format" test, which verified the full format, and then have all the type conversion tests verify just that details['owner_name']['data_type'] == <expected_type>